### ISSUE TITLE
"input" does not render API documentation for FormType with DataTransformer - No Errors Thrown ...

### ISSUE THREAD

**Issue by bzitzow** [nelmio/NelmioApiDocBundle#208]
### Use Case

I have a FormType that requires a user's email address, which is in turn is transformed via a DataTransformer into a User object so it may be bound to the form's data class as a related User. Although the FormType named 'email' is a 'text' FormType, no documentation is generated for this form. 

If this requires an enhancement, I'm more than happy to help - however I may need some initial direction. 

Also, This is my first time using this bundle, so if I have misunderstood how it is intended to work in this scenario some constructive feedback would be appreciated. 

Thank you!

Composer.json:

```
"require": {
    "symfony/symfony": "2.3.*",
    "jms/serializer-bundle": "0.12.*@dev",
    "friendsofsymfony/rest-bundle": "0.13.*@dev",
    "nelmio/api-doc-bundle": "2.3.*@dev"
},
```

The Controller:

```
/**
 * @ApiDoc(
 *      input="Vendor\MainBundle\Form\SystemType"
 * )
 */
public function postSystemAction(Request $request)
{
    $form = $this->createForm(
        new SystemType(),    // FormType
        new System(),                // Entity
        array('entityManager' => $this->getDoctrine()->getManager()) // options
    );

    // Process Stuff ;)
}
```

The FormType:

```
class SystemType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        $transformer = new EmailToUserTransformer($options['entityManager']);

        $builder
            ->add(
                $builder->create('email', 'text', array('property_path' => 'user'))
                ->addViewTransformer($transformer)
            )
            ->add('cpuid', 'text', array('required' => true, 'description' => 'The CPU id'))
            ->add('mac_address', 'text', array('required' => true, 'property_path' => 'macAddress'))
        ;
    }

    public function setDefaultOptions(OptionsResolverInterface $resolver)
    {
        $resolver->setDefaults(array(
            'data_class'        => 'Vendor\MainBundle\Entity\User\System',
            'csrf_protection'   => false
        ));

        $resolver->setRequired(array(
            'entityManager',
        ));

        $resolver->setAllowedTypes(array(
            'entityManager' => 'Doctrine\Common\Persistence\ObjectManager',
        ));
    }

    public function getName()
    {
        return 'vendor_mainbundle_systemtype';
    }
}
```

**Comment by nazar256** [comment]
Hi guys, I've faced the same problem. But looking at how ApiDoc bundle works for few minutes I've found a solution for my case! Here it is:
1. First, define your form as service, it is [pretty easy](http://symfony.com/doc/current/book/forms.html#defining-your-forms-as-services), suppose it is "form_alias_you_defined"
2. Inject needed services to form (ex. EntityManager)
3. Use your form name in ApiDoc like 
```php
/**
* ...
* input = form_alias_you_defined
```

Profit!
Also don't forget to read official documentation with usefull tips, like passing method (GET, POST, etc).
